dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task210702UpdateStructureTable.java

```diff
@@ -0,0 +1,123 @@
+package com.dotmarketing.startup.runonce;
+
+import com.dotmarketing.common.db.DotDatabaseMetaData;
+import com.dotmarketing.startup.AbstractJDBCStartupTask;
+import com.dotmarketing.util.Logger;
+import java.sql.SQLException;
+import java.util.List;
+
+/**
+ * Upgrade task used to add columns `icon` and `sort_order` in `structure` table if needed
+ */
+public class Task210702UpdateStructureTable extends AbstractJDBCStartupTask {
+
+    private final DotDatabaseMetaData dotDatabaseMetaData = new DotDatabaseMetaData();
+
+    @Override
+    public boolean forceRun() {
+        try {
+            return !dotDatabaseMetaData.hasColumn("structure", "icon") &&
+                    !dotDatabaseMetaData.hasColumn("structure", "sort_order") ;
+        } catch (SQLException e) {
+            Logger.error(this, e.getMessage(),e);
+            return false;
+        }
+    }
+
+    /**
+     * Returns the upgrade SQL query for PostgreSQL.
+     *
+     * @return The SQL statement(s).
+     */
+    @Override
+    public String getPostgresScript() {
+        String query = "";
+        try {
+            if(!dotDatabaseMetaData.hasColumn("structure", "icon")) {
```

Comment:
Isn't this redundant since the `forceRun` already guarantees it?

Reply:
No, b/c the `forceRun` guarantees that both columns exist, if only one exists we need to check which one to create the other one.

Reply:
```public boolean forceRun() {
        try {
            return !dotDatabaseMetaData.hasColumn("structure", "icon") &&
                    !dotDatabaseMetaData.hasColumn("structure", "sort_order") ;
        } catch (SQLException e) {
            Logger.error(this, e.getMessage(),e);
            return false;
        }
    }
```
that looks to me that forceRun true = neither exist